packages/react/src/utils/useEventCallback.ts

```diff
@@ -1,4 +1,44 @@
 'use client';
-import { useEffectEvent } from '@floating-ui/react/utils';
```

Comment:
For reference:

[Source link](https://github.com/floating-ui/floating-ui/blob/ae1329d3173f8fd57c3a8808c6d1ae6a5e6d7825/packages/react/src/utils/hooks.ts#L17-L38)

```typescript
const useInsertionEffect = SafeReact.useInsertionEffect as
  | AnyFunction
  | undefined;

const useSafeInsertionEffect = useInsertionEffect || ((fn) => fn());

export function useEffectEvent<T extends AnyFunction>(callback?: T) {
  const ref = React.useRef<AnyFunction | undefined>(() => {
    if (__DEV__) {
      throw new Error('Cannot call an event handler while rendering.');
    }
  });

  useSafeInsertionEffect(() => {
    ref.current = callback;
  });

  return React.useCallback<AnyFunction>(
    (...args) => ref.current?.(...args),
    [],
  ) as T;
}
```